### PR TITLE
Enable `optimize_syscalls` when `MAIN_MODULE=2`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
With `MAIN_MODULE=1` we cannot disable `SYSCALLS_REQUIRE_FILESYSTEM` since the side modules might depend on these syscalls.  However we `MAIN_MODULE=2` we have a complete view of the side module imports so we can.

Also, rename `optimize_syscalls` so it better describes what it does.

### CODE REVIEW

File: test/other/metadce/test_metadce_hello_dylink.jssize [hunk 1]
```diff
@@ -1 +1 @@
-28024
+15091
```
Comment (kripken) [inline test/other/metadce/test_metadce_hello_dylink.jssize:1]: Nice!

**Review by kripken — COMMENTED** [review]
Great!

This is covered by existing tests I guess?

**Comment by sbc100** [comment]
> Great!
> 
> This is covered by existing tests I guess?

We have a lot of `-sMAIN_MODULE=2` tests yes, and I believe many of them do filesystem stuff.

**Comment by sbc100** [comment]
OK to land?

**Review by kripken — APPROVED** [review]
Sorry, I thought I lgtm'd before... lgtm!

**Pull request merged**